lib/ansible/modules/cloud/amazon/ec2_ami_facts.py

```diff
@@ -189,18 +189,43 @@
 def list_ec2_images(ec2_client, module):
 
     image_ids = module.params.get("image_ids")
-    filters = ansible_dict_to_boto3_filter_list(module.params.get("filters"))
     owners = module.params.get("owners")
     executable_users = module.params.get("executable_users")
+    filters = module.params.get("filters")
+    owner_param = []
+
+    # describe_images is *very* slow if you pass the `Owners`
+    # param (unless it's self), for some reason.
+    # Converting the owners to filters and removing from the
+    # owners param greatly speeds things up.
+    # Implementation based on aeiou's suggestion in #24886
```

Comment:
aioue ;)

Reply:
Oops, I should have checked rather than trying to do it from memory :). Fixed!